src/main/java/com/lineate/testyourlexicon/services/implementations/UserServiceImpl.java

```diff
@@ -0,0 +1,50 @@
+package com.lineate.testyourlexicon.services.implementations;
+
+import com.lineate.testyourlexicon.dto.UserRegistrationDto;
+import com.lineate.testyourlexicon.entities.User;
+import com.lineate.testyourlexicon.repositories.UserRepository;
+import com.lineate.testyourlexicon.services.UserRegistrationResponse;
+import com.lineate.testyourlexicon.services.UserService;
+import lombok.AllArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class UserServiceImpl implements UserService {
+
+  private final UserRepository userRepository;
+  private final PasswordEncoder passwordEncoder;
+  private final Logger logger = LoggerFactory.getLogger(UserServiceImpl.class);
+
+  private void logUserRegistration(UserRegistrationDto userRegistrationDto) {
+    logger.info("Registered user: {first name: {}, last name: {}, email: {}}",
+      userRegistrationDto.getFirstName().toLowerCase(),
+      userRegistrationDto.getLastName().toLowerCase(),
+      userRegistrationDto.getEmail());
+  }
+
+  public UserRegistrationResponse createUser(UserRegistrationDto userRegistrationDto) {
```

Comment:
Service should return either entity objects or Dtos. Let's create a Dto class for User and return an instance of UserDto here.